File: src/python/_1161_Maximum_Level_Sum_of_a_Binary_Tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def maxLevelSum(self, root: Optional[TreeNode]) -> int:
        dq = deque([root] if root else [])
        maxLevel = currLevel = 1
        maxSum = float("-inf")

        while dq:
            currSum = 0
            for _ in range(len(dq)):
                node = dq.popleft()
                currSum += node.val
                if node.left:
                    dq.append(node.left)
                if node.right:
                    dq.append(node.right)

            if currSum > maxSum:
                maxSum = currSum
                maxLevel = currLevel

            currLevel += 1

        return maxLevel

File: src/java/_1161_Maximum_Level_Sum_of_a_Binary_Tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 * int val;
 * TreeNode left;
 * TreeNode right;
 * TreeNode() {}
 * TreeNode(int val) { this.val = val; }
 * TreeNode(int val, TreeNode left, TreeNode right) {
 * this.val = val;
 * this.left = left;
 * this.right = right;
 * }
 * }
 */
class Solution {
    public int maxLevelSum(TreeNode root) {
        int maxLevel = 1, maxSum = Integer.MIN_VALUE, currLevel = 1;
        Deque<TreeNode> dq = new ArrayDeque<>();

        dq.add(root);

        while (dq.size() > 0) {
            int size = dq.size(), currSum = 0;
            for (int i = 0; i < size; i++) {
                TreeNode node = dq.removeFirst();
                currSum += node.val;
                if (node.left != null)
                    dq.add(node.left);
                if (node.right != null)
                    dq.add(node.right);
            }
            if (currSum > maxSum) {
                maxSum = currSum;
                maxLevel = currLevel;
            }
            currLevel += 1;
        }

        return maxLevel;
    }
}
